src/Form/AuthenticationForm.php

```diff
@@ -326,12 +346,26 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
   public function submitForm(array &$form, FormStateInterface $form_state) {
     // Get the processed value from the form state.
     $processed_submitted = $form_state->get('processed_submitted');
+    $active_key = $this->activeKey;
 
-    if (!empty($processed_submitted) && $this->keyIsWritable($this->activeKey)) {
+    if (!empty($processed_submitted) && $this->keyIsWritable($active_key)) {
```

Comment:
It seems the environment variables key provider does not work properly because the class `EnvironmentVariablesKeyProvider` implements the interface `KeyProviderSettableValueInterface`. Because it is checked whether the provider class implements that interface, the key is not loaded correctly so the connection cannot be tested using the "Send request" button.
It works correctly on my local development environment if I remove that interface name from the interface list and the functions `setKeyValue()` and `deleteKeyValue()` in the class `EnvironmentVariablesKeyProvider`.

Reply:
@Jaesin Could you fix this one too? Environment variable key storage provided by the module should work properly.

Reply:
This brings up the question why is `EnvironmentVariablesKeyProvider` settable?

Reply:
This has been addressed by making sure `keyIsWritable()` checks the input type isn't `none`.